s3util/uploader.go

```diff
@@ -199,6 +199,9 @@ func (u *uploader) putPart(p *part) error {
 }
 
 func (u *uploader) Close() error {
+	if u == nil {
```

Comment:
usually we check u itself to be nil but this works as well.  Not sure which method is idiomatic.